lwk/lwkwallet.go

```diff
@@ -0,0 +1,202 @@
+	ctx := context.Background()
+	fundedTx, err := r.lwkClient.send(ctx, &sendRequest{
+		Addressees: []*unvalidatedAddressee{
+			{
+				Address: swapParams.OpeningAddress,
+				Satoshi: swapParams.Amount,
+			},
+		},
+		WalletName: r.walletName,
+	})
+	if err != nil {
+		return "", "", 0, err
+	}
+	signed, err := r.lwkClient.sign(ctx, &signRequest{
+		SignerName: r.signerName,
+		Pset:       fundedTx.Pset,
+	})
+	if err != nil {
+		return "", "", 0, err
+	}
+	broadcasted, err := r.lwkClient.broadcast(ctx, &broadcastRequest{
+		WalletName: r.walletName,
+		Pset:       signed.Pset,
+	})
+	if err != nil {
+		return "", "", 0, err
+	}
+	p, err := psetv2.NewPsetFromBase64(signed.Pset)
+	if err != nil {
+		return "", "", 0, err
+	}
+	err = psetv2.FinalizeAll(p)
+	if err != nil {
+		return "", "", 0, err
+	}
+	tx, err := psetv2.Extract(p)
+	if err != nil {
+		return "", "", 0, err
+	}
+	txhex, err := tx.ToHex()
+	if err != nil {
+		return "", "", 0, err
+	}
+	return broadcasted.Txid, txhex, 0, nil
+}
+
+// GetBalance returns the balance in sats
+func (r *LWKRpcWallet) GetBalance() (uint64, error) {
+	ctx := context.Background()
+	balance, err := r.lwkClient.balance(ctx, &balanceRequest{
+		WalletName: r.walletName,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return uint64(balance.Balance[network.Regtest.AssetID]), nil
+}
+
+// GetAddress returns a new blech32 address
+func (r *LWKRpcWallet) GetAddress() (string, error) {
+	ctx := context.Background()
+	address, err := r.lwkClient.address(ctx, &addressRequest{
+		WalletName: r.walletName})
+	if err != nil {
+		return "", err
+	}
+	return address.Address, nil
+}
+
+// SendToAddress sends an amount to an address
+func (r *LWKRpcWallet) SendToAddress(address string, amount uint64) (string, error) {
+	ctx := context.Background()
+	sendres, err := r.lwkClient.send(ctx, &sendRequest{
+		WalletName: r.walletName,
+		Addressees: []*unvalidatedAddressee{
+			{
+				Address: address,
+				Satoshi: amount,
+			},
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	signed, err := r.lwkClient.sign(ctx, &signRequest{
+		SignerName: r.signerName,
+		Pset:       sendres.Pset,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	broadcastres, err := r.lwkClient.broadcast(ctx, &broadcastRequest{
+		WalletName: r.walletName,
+		Pset:       signed.Pset,
+	})
+	if err != nil {
+		return "", err
+	}
+	return broadcastres.Txid, nil
+}
+
+func (r *LWKRpcWallet) SendRawTx(txHex string) (string, error) {
+	ctx := context.Background()
+	res, err := r.electrumClient.BroadcastTransaction(ctx, txHex)
+	if err != nil {
+		return "", err
+	}
+	return res, nil
+}
+
+func (r *LWKRpcWallet) GetFee(txSize int64) (uint64, error) {
+	ctx := context.Background()
+	feeRes, err := r.electrumClient.GetFee(ctx, wallet.LiquidTargetBlocks)
+
+	satPerByte := float64(feeRes) / float64(kiloByte)
+	if satPerByte < minimumSatPerByte {
+		satPerByte = minimumSatPerByte
+	}
+	if err != nil {
+		satPerByte = minimumSatPerByte
+	}
```

Comment:
A log line in case of an error could be helpful for debugging purposes in the future.